### PR TITLE
Lint that DSA is not used - BR 

### LINKED ISSUE #574: Lint that DSA is not used - BR
This issue duplicates-ish #537 

A lint was implemented in #572 which checks this against the Mozilla effective date. This issue is to implement a second which tracks against the BR effective date.

### PR DESCRIPTION
**Pull request by christopher-henderson**
I admit that this was largely a copy-paste job of #572 (thanks @robplee !) so right there might be an obvious source of mistakes.

@sleevi I believe this got to at least the minimum of what you were referring to in #537 (sunsetting `dh_params_missing` because it simply doesn't apply anymore). Although, it sounded as though that there might be a slew of DSA BR lints that I may have to hunt down and sunset.

This PR resolves #574

### CODE REVIEW

**Comment by christopher-henderson** [comment]
Ugh, I think the `golangci` tool is looking at all touched files and it doesn't like that we're importing the deprecated `crypto/dsa` package....

**Review by robplee — APPROVED** [review]
Looks good to me, although unless I'm missing something you've not added an import to crypto/dsa so could you explain what you meant about that in your previous comment on the PR?

**Review by cpu — CHANGES_REQUESTED** [review]
Thanks! Looks good to me once the Golang CI config change is rolled back. We can either administratively merge or wait until master is fixed to merge.

File: .golangci.yaml
Comment (christopher-henderson) [reply .golangci.yaml:36]: Yup, I actually wanted to block my own review because of this but apparently you can't do that.

Comment (christopher-henderson) [reply .golangci.yaml]: Alright, this all straightened out now. It should be the lone lint now.

**Review by cpu — APPROVED** [review]
Thanks!

**Pull request merged**
